examples/testdata/wasi_args.go

```diff
@@ -0,0 +1,53 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"syscall"
+)
+
+//go:wasm-module wasi_snapshot_preview1
+//export args_get
+func args_get(uint32, uint32) uint16
+
+//go:wasm-module wasi_snapshot_preview1
+//export args_sizes_get
+func args_sizes_get(uint32, uint32) uint16
+func main() {
+	testGetWASIArgs()
+	testInvalidArguments()
+}
+
+func testGetWASIArgs() {
+	var expectedArgs = []string{
+		"foo",
+		"bar",
+		"foobar",
+		"",
+		"baz",
+	}
+
+	if len(os.Args) != len(expectedArgs) {
+		panic(fmt.Sprintf("The length of the args is not expected: %v", len(os.Args)))
+	}
+
+	for i, expectedArg := range expectedArgs {
+		if os.Args[i] != expectedArg {
+			panic(fmt.Sprintf("os.Args are not what are expected. expected: %v, actual: %v", expectedArgs, os.Args))
+		}
+	}
+}
+
+func testInvalidArguments() {
```

Comment:
this sort of test I think we can handle using the text format and not in an example. The less we do here the less scary the example is. If we aren't confident about something the example relies on being tested in tinygo, we can raise a PR there to make it better.

Reply:
sorry, could you elaborate what made you feel this test validates tinyGo, not the implementation / spec of the WASI system call? It's testing that WASI's argument AP's behavior

Reply:
sorry it was the above function (misplaced comment) which was looking at `os.Args`. here really it is most direct to just export the function and test it vs trying to invoke test code in wasm